internal/ir/builder.go

```diff
@@ -624,6 +624,12 @@ func (b *builder) expr(fn *Function, e ast.Expr, expand bool) Value {
 		value = b.binaryExpr(fn, expr)
 	case *ast.ObjectExpr:
 		value = b.objectExpr(fn, expr)
+	case *ast.SubscriptExpr:
+		value = &SubscriptExpr{
+			node:   node{expr},
+			Object: b.expr(fn, expr.Object, false /* expand */),
```

Comment:
We should expand the object value, since we can have this code `[10,20,30][1]` the object list should be converted into a temporary variable.

We should expand expand the index too, since we can have complex index values (not only Const or Var).